### PR TITLE
Allow the user to select multiple images in the Media Library

### LINKED ISSUE #35181: [Enhancement]: Allow the user to select multiple images in the Media Library
### Describe the solution you'd like

We currently only allow the user to select one image to add to the product gallery, but the user should be able to select any number of files.

**Acceptance criteria**

- [ ] The user can select multiple images in the Media Library, then add them to the image gallery in the product form.
- [ ] The same applies to the system file picker while choosing images to upload.

### Describe alternatives you've considered

_No response_

### Additional context

This is how it currently works in the gallery block in Gutenberg.

### PR DESCRIPTION
**Pull request by octaedro**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR adds the code to allow users to select multiple images in the Media Library.

Screenshot 2022-11-25 at 10 02 22

Closes #35181.

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

1. Checkout this branch and turn on the `new-product-management-experience`.
2. Go to `Products` > `Add New (MVP)` > `Images`.
3. Press `Choose images` and select more than one image (by pressing `command` and selecting them).
4. The selected images should be added to the gallery in the image section.
5. Repeated images won't be added.
6. Verify that the storybook works correctly.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce-admin/client/products/sections/images-section.tsx [hunk 1]
```diff
@@ -174,19 +175,22 @@ export const ImagesSection: React.FC = () => {
...
+									onSelect={ ( files ) => {
 										if (
-											images.find(
-												( img ) => file.id === img.id
-											) === undefined
+											intersectionBy(
+												images,
+												files,
+												'id'
+											).length === 0
```
Comment (louwie17) [inline plugins/woocommerce-admin/client/products/sections/images-section.tsx]: This prevents any image from being added as soon as one duplicate image is part of the list.
Could we just filter out the duplicate images? and add all the newly selected images?

**Review by louwie17 — COMMENTED** [review]
This overall tested well @octaedro, I did notice an issue when I tried to add several images were one of them would already be in the list. This prevented all the selected images from being added. Could we just filter out the already selected images?

File: plugins/woocommerce-admin/client/products/sections/images-section.tsx (hunk 1 above)
Comment (octaedro) [reply plugins/woocommerce-admin/client/products/sections/images-section.tsx]: Good idea! I fixed that in the commit [9a29f9c](https://github.com/woocommerce/woocommerce/pull/35722/commits/9a29f9ce1d073ccfaa8bd78702ab562b80e77411)

**Comment by octaedro** [comment]
Thank you @louwie17 for your review.
I just addressed the change you suggested. Could you take another look at this PR?

**Review by louwie17 — APPROVED** [review]
Thanks for fixing the repeated images issue, nice work!
This is still testing well, LGTM 🚀

**Pull request merged**